src/plugins/intel_gna/src/scale_factor_helper.cpp

```diff
@@ -0,0 +1,113 @@
+            inputs.Get().at(id).scale_factor = input_scale_factors[id];
+        }
+    }
+}
+
+static bool IsCustomInputScaleFactorPerInputAvailable(const std::map<std::string, float>& per_input_scale_factors) {
+    if (per_input_scale_factors.empty()) {
+        return false;
+    }
+
+    bool is_scale_factor_custom = false;
+    for (const auto& scale_factor : per_input_scale_factors) {
+        if (!fp32eq(scale_factor.second, GNAPluginNS::kScaleFactorDefault)) {
+            is_scale_factor_custom = true;
+            break;
+        }
+    }
+
+    return is_scale_factor_custom;
+}
+
+static void ApplyScaleFactorsPerInput(const std::map<std::string, float>& per_input_scale_factors,
+                                      GNAPluginNS::GnaInputs& inputs) {
+    if (per_input_scale_factors.size() > inputs.size()) {
+        IE_THROW() << "Configuration per input scale factors count is bigger than inputs count";
+    }
+
+    for (auto&& sf : per_input_scale_factors) {
+        if (!fp32eq(sf.second, GNAPluginNS::kScaleFactorDefault)) {
+            auto input_it = inputs.find(sf.first);
+            if (input_it == inputs.end()) {
+                IE_THROW() << "Given scale factor for invalid input: " << sf.first;
+            }
+            ::log::debug() << "Using input scale factor defined in configuration for input " << sf.first << std::endl;
+            input_it->scale_factor = sf.second;
+        }
+    }
+}
+
+static bool CheckIfCanApplyCustomScaleFactor(const GNAPluginNS::HeaderLatest::ModelHeader& header) {
+    return header.version.major <= 2 ? header.version.minor <= 7 : false;
+}
+
+void ApplyInputScaleFactors(const GNAPluginNS::Config& config,
+                            const GNAPluginNS::HeaderLatest::ModelHeader& header,
+                            GNAPluginNS::GnaInputs& inputs) {
+    // If scale factors are defined in configuration we still need to use them instead of imported values,
+    // for example to change the scale factors for the old models.
+
+    bool custom_scale_factor_per_input = IsCustomInputScaleFactorPerInputAvailable(config.inputScaleFactorsPerInput);
+    bool custom_scale_factor_legacy = IsCustomInputScaleFactorAvailableLegacy(config.inputScaleFactors);
+
+    if (!custom_scale_factor_per_input && !custom_scale_factor_legacy) {
+        return;
+    }
+
+    if (!CheckIfCanApplyCustomScaleFactor(header)) {
+        ::log::warning() << "Cannot apply custom scale factor for model versions >= 2.7\n";
+        return;
+    }
+
+    // Due the fact inputScaleFactors is set by defuault construcor of GNAPluginNS::Config
+    // we need to check is_intput_scale_factor_per_input_given as first.
+    if (custom_scale_factor_per_input) {
+        ApplyScaleFactorsPerInput(config.inputScaleFactorsPerInput, inputs);
+        return;
+    }
+
+    ApplyScaleFactorsLegacy(config.inputScaleFactors, inputs);
```

Comment:
```suggestion
if(IsCustomInputScaleFactorAvailableLegacy(config.inputScaleFactors)) {
        ApplyScaleFactorsLegacy(config.inputScaleFactors, inputs);
    }
```

Reply:
it is needed to avoid redundant warning logs